ckanext/recombinant/views.py

```diff
@@ -374,25 +370,26 @@ def _schema_json(dataset_type, published_resource=False):
             fld = OrderedDict()
             resource['fields'].append(fld)
             fld['id'] = field['datastore_id']
-            if fld['id'] in pkeys:
-                fld['obligation'] = 'mandatory'
-            elif field.get('excel_required'):
-                fld['obligation'] = 'mandatory'
-            elif field.get('excel_required_formula'):
-                fld['obligation'] = 'conditional'
-            else:
-                fld['obligation'] = 'optional'
+            fld['obligation'] = OrderedDict()
+            for lang in config['ckan.locales_offered'].split():
+                with force_locale(lang):
+                    if fld['id'] in pkeys:
+                        fld['obligation'][lang] = _('Mandatory')
+                    elif field.get('excel_required'):
+                        fld['obligation'][lang] = _('Mandatory')
+                    elif field.get('excel_required_formula'):
+                        fld['obligation'][lang] = _('Conditional')
+                    else:
+                        fld['obligation'][lang] = _('Optional')
```

Comment:
because obligation is a controlled list it might be nicer to have the code value instead of translated text in the json

Reply:
@wardi I also thought that was the case with `obligation` (and maybe it was at some point). But `obligation` is not used in any of the logic anywhere. It was only outputted in the Reference sheet for the XLSX files. I have since outputted it in the DataDictionaries. 

It is strictly a descriptor to a field and is only shown in Reference sheet and DataDictionary. E.g.
```
obligation:
      en: This field is mandatory if there is an amount reported under “Number of applications submitted via other channels”
      fr: Ce champ est requis s’il y a un montant rapporté sous le champ « Nombre de demandes soumises par les autres modes de prestation »
```

Reply:
Where does this code copy the obligation dict if already defined in en/fr in the yaml?

Reply:
it does it underneath the above, with the `label, description, validation`